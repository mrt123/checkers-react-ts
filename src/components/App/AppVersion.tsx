import styled from "styled-components";

const AppVersionContainer = styled.div`
  position: absolute;
  font-size: 10px;
  bottom: 10px;
  left: 10px;
`;

const AppVersion = () => {
  return (
    <AppVersionContainer>
      v: {process.env.REACT_APP_VERSION}
    </AppVersionContainer>
  );
};

export default AppVersion;
