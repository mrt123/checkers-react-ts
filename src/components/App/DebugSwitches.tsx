import { FormGroup } from "@mui/material";
import DebugSwitch from "../general/DebugSwitch";

const DebugSwitches = () => {
  return (
    <FormGroup>
      <DebugSwitch stateAttribute="reduxLogs" label="redux logs" />
      <DebugSwitch
        stateAttribute="fieldsCoordinates"
        label="fields coordinates"
      />
    </FormGroup>
  );
};

export default DebugSwitches;
