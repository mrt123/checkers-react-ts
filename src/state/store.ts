import { applyMiddleware, createStore } from "redux";
import { createLogger } from "redux-logger";
import {
  SET_ACTIVE_PIN,
  UNSET_HIGHLIGHT_FIELD,
} from "../components/game/Board/state/boardActions";
import rootReducer from "./reducers";

const middleWares = [];

if (window.location.hostname === "localhost") {
  const loggerMiddleware = createLogger({
    collapsed: true,
    predicate: (getState, action) => {
      const actionIsUnsettingActivePin =
        action.type === SET_ACTIVE_PIN && action.pin === null;

      return (
        !actionIsUnsettingActivePin &&
        action.type !== UNSET_HIGHLIGHT_FIELD &&
        true
        // getState().debugSwitches.reduxLogs
      );
    },
  });
  middleWares.push(loggerMiddleware);
}

const store = createStore(rootReducer, applyMiddleware(...middleWares));
export type RootState = ReturnType<typeof store.getState>;
export default store;
