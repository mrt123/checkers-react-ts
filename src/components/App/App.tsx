import FormGroup from "@mui/material/FormGroup";
import { useSelector } from "react-redux";
import styled from "styled-components";
import { RootState } from "../../state/store";
import Board from "../game/Board/Board";
import { Field } from "../game/Board/state/boardStateTypes";
import MoveHistory from "../game/MoveHistory/MoveHistory";
import PlayerColorIndicator from "../game/Pin/PlayerColorIndicator";
import DebugSwitch from "../general/DebugSwitch";
import AppVersion from "./AppVersion";

const AppContainer = styled.div`
   {
    display: flex;
    justify-content: space-evenly;
    background-color: #282c34;
    min-height: 100vh;
    font-size: calc(10px + 2vmin);
    color: white;
    padding-top: 30px;
  }
`;

const InfoIcon = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 15px;
  padding: 10px;
`;

const InfoText = styled.div`
  margin: 10px;
`;

const App = () => {
  const board = useSelector((state: RootState) => state.board);

  const pinColorsLeft = board.fields.reduce(
    (accumulator: string[], field: Field) => {
      const pinColor = field.pin?.color;
      if (pinColor) {
        if (!accumulator.includes(pinColor)) accumulator.push(pinColor);
      }
      return accumulator;
    },
    []
  );

  const gameIsOver = pinColorsLeft.length === 1;
  const playerInfoColour = gameIsOver
    ? pinColorsLeft[0]
    : board.activePlayer.color;
  const playerInfoText = gameIsOver ? "wins" : "next";

  // TODO: when playing against computer, generate alll possible moves, and choose one randomly

  const isLocalshost = window.location.hostname === "localhost";
  const debuSwitches = !isLocalshost ? null : (
    <FormGroup>
      <DebugSwitch stateAttribute="reduxLogs" label="redux logs" />
      <DebugSwitch
        stateAttribute="fieldsCoordinates"
        label="fields coordinates"
      />
    </FormGroup>
  );

  return (
    <AppContainer>
      <div>
        <Board />
        <InfoIcon>
          <PlayerColorIndicator color={playerInfoColour} />
          <InfoText>{playerInfoText}</InfoText>
        </InfoIcon>
        <AppVersion />
      </div>
      <div>
        <MoveHistory />
        {debuSwitches}
      </div>
    </AppContainer>
  );
};

export default App;
