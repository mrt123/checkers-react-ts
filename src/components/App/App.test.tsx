import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../../state/store";
import App from "./App";

window.HTMLElement.prototype.scrollIntoView = jest.fn();

test("matches snapshot", () => {
  const { container } = render(
    <Provider store={store}>
      <App />
    </Provider>
  );
  expect(container).toMatchSnapshot();
});
