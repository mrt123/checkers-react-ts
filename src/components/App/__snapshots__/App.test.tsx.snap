// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="sc-bqWxrE laMVq"
  >
    <div>
      <div
        class="sc-jSUZER aqICG"
      >
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        >
          <div
            class="sc-bcXHqe gGFMfw react-draggable"
            color="rgb(222,222,222)"
            style="transform: translate(0px,0px);"
          >
            <div
              class="sc-dkrFOg fmVpSy"
            >
              <div
                class="sc-gswNZR eMwWNB"
                direction="up"
              />
            </div>
          </div>
        </div>
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        >
          <div
            class="sc-bcXHqe fKBXoM react-draggable"
            color="orange"
            style="transform: translate(0px,0px);"
          >
            <div
              class="sc-dkrFOg fmVpSy"
            >
              <div
                class="sc-gswNZR ijbjVD"
                direction="down"
              />
            </div>
          </div>
        </div>
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
        <div
          class="sc-hLBbgP dCxjxY"
          color="black"
        />
        <div
          class="sc-hLBbgP bldSqn"
          color="white"
        />
      </div>
      <div
        class="sc-ipEyDJ iwjXVa"
      >
        <span
          class="sc-gKPRtg gXNtLV"
          color="white"
        />
        <div
          class="sc-csuSiG bLCRUo"
        >
          next
        </div>
      </div>
      <div
        class="sc-eDWCr hsPPCD"
      >
        v: 
        1.1.0
      </div>
    </div>
    <div>
      <div
        class="sc-kDvujY icIOHb"
      >
        <div />
      </div>
      <div
        class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
      >
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-1f2kvjf-MuiFormControlLabel-root"
        >
          <span
            class="MuiSwitch-root MuiSwitch-sizeSmall css-1ycxvqq-MuiSwitch-root"
          >
            <span
              class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
            >
              <input
                class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                type="checkbox"
              />
              <span
                class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
            />
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
          >
            reduxLogs
          </span>
        </label>
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-1f2kvjf-MuiFormControlLabel-root"
        >
          <span
            class="MuiSwitch-root MuiSwitch-sizeSmall css-1ycxvqq-MuiSwitch-root"
          >
            <span
              class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
            >
              <input
                class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                type="checkbox"
              />
              <span
                class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
            />
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
          >
            fieldsCoordinates
          </span>
        </label>
      </div>
    </div>
  </div>
</div>
`;
